apps/google-analytics-4/frontend/src/locations/Sidebar.tsx

```diff
@@ -1,7 +1,21 @@
 import AnalyticsApp from 'components/main-app/analytics-app/AnalyticsApp';
+import useKeyService from 'hooks/useKeyService';
+
+// TO DO: Add propertyId and reportSlug logic
 
 const Sidebar = () => {
-  return <AnalyticsApp />;
+  const { parameters } = useKeyService({});
```

Comment:
Hm, I think we should _not_ get the parameters from the `useKeyService` and instead just fetch them directly from the app sdk. Then you can avoid all those refactorings you're needing to do in that hook.